### PR TITLE
Fixed the timeout issue when retryCount > 0 on the taskdefinition

### PR DESCRIPTION
**Pull request by jvemugunta**
This is in reference to the issue #780

@v1r3n and @cyzhao - Can you please review this PR when you get a chance

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 1]
```diff
@@ -99,7 +101,7 @@
 
     @Inject
     public WorkflowExecutor(DeciderService deciderService, MetadataDAO metadataDAO, ExecutionDAO executionDAO,
-                            QueueDAO queueDAO, ParametersUtils parametersUtils, Configuration config) {
+            QueueDAO queueDAO, ParametersUtils parametersUtils, Configuration config) {
```
Comment (cyzhao) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: please avoid format changes in a bug/feature PR, it makes it harder to review.

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 2]
```diff
@@ -240,9 +251,16 @@ public void rewind(String workflowId) {
         }
 
         WorkflowDef workflowDef = metadataDAO.get(workflow.getWorkflowType(), workflow.getVersion());
-        if (!workflowDef.isRestartable() && workflow.getStatus().equals(WorkflowStatus.COMPLETED)) { // Can only restart non completed workflows when the configuration is set to false
-            throw new ApplicationException(CONFLICT, String.format("WorkflowId: %s is an instance of WorkflowDef: %s and version: %d and is non restartable",
-                    workflowId, workflowDef.getName(), workflowDef.getVersion()));
+        if (!workflowDef.isRestartable() && workflow.getStatus().equals(WorkflowStatus.COMPLETED)) { // Can only restart
+                                                                                                     // non completed
```
Comment (cyzhao) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: This format change is completely off.

**Comment by cyzhao** [comment]
@jvemugunta can you revert format changes, or mark code changes related to the fix?

**Comment by jvemugunta** [comment]
I am using Eclipse as an IDE and I am not sure how the formatting is the changing. Is there a . settings we can use for all developers who are using eclipse. This way the indentation and formatting is consistent for all developers

**Comment by panthony** [comment]
@jvemugunta Long time since I did use eclipse but I believe you can at least disable the "auto formatting" when saving a file.

The most cross-IDE way to have a common formatting is to use `.editorconfig` (https://editorconfig.org/) which is understood by a lot of IDEs

**Comment by jvemugunta** [comment]
@cyzhao - These are the changes made. I have changed my IDE to IntelliJ . I am hoping the formatting to match with the upstream/dev branch

https://github.com/jvemugunta/conductor/blob/174871f348a088c2fe4e7c3bb264c1e076ad6f74/core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java#L618-L622

https://github.com/jvemugunta/conductor/blob/174871f348a088c2fe4e7c3bb264c1e076ad6f74/server/src/main/java/com/netflix/conductor/server/ServerModule.java#L120

**Comment by jvemugunta** [comment]
@cyzhao - This changes are done.Please review the PR

**Comment by cyzhao** [comment]
@jvemugunta changes look good to me, please resolve the conflict with ServerModule.java before we can merge.

**Comment by jvemugunta** [comment]
@cyzhao  - I have added wrong snippet of code.
Can you reopen the PR and I will add the right one

**Comment by cyzhao** [comment]
@jvemugunta merged #800

**Comment by JasperB** [comment]
@cyzhao 
we also met this problem the task status marked as 'IN_PROGRESS' and not execute anymore, I saw this issue here.
So I want ask the question is @jvemugunta applied the changes with two classes change, but I saw only the WorkflowExecutor.java merged to release1.12.3 branch, we met that issue may caused the other class(ServerModel.java) not merged to this release branch?

**Pull request merged**